### PR TITLE
Add Namespace::Index::Cooldown for namespaced cooldown

### PR DESCRIPTION
**Pull request by kaspth**
Extracts a Namespace::Index::Manifest to hold the /versions contents that we can append and compact continually.

This means that a Namespace::Index and Namespace::Index::Cooldown can choose when to append a "new" version: immediately or delayed by the cooldown interval.

We can then independently `compact` these manifests. That part still feels a little obscure and brittle in how we gather the "composed" envelopes.

---

This also still means that we need the shallow clone stuff from #165 and `Namespace::Index::Mirror`. This code assumes we can gather relatively accurate `published_at` traits. Probably fine if we're off by some hours.

- [x] Convert `Version#line` to a column
- [x] Remove `Namespace::Gem::Info` and table
- [x] Remove `last_compacted_at` and `versions_contents` from `Namespace::Index`.
- [x] Tests besides the manual ones I did.

### CODE REVIEW

**Comment by indirect** [comment]
I like this approach! Maybe we can port over the mirror class to know which gems to import, and the cooldown tests to make sure the edge cases are handled? Happy to work together on that if you like.

**Comment by kaspth** [comment]
> I like this approach! Maybe we can port over the mirror class to know which gems to import, and the cooldown tests to make sure the edge cases are handled? Happy to work together on that if you like.

awesome, yeah that sounds good!

**Comment by kaspth** [comment]
@indirect actually how do you feel about adding Mirror separately? I don't think it's integral to this since I'm not removing CooldownVersion in this PR. I'll try to port the tests over.

**Comment by kaspth** [comment]
ok looks like the tests in #165 are mostly testing the Namespace::Mirror, though there might be some cooldown tests in there too. anyway, I added some distinct cooldown tests now. I need to do some more additions, but I have to run now.

**Comment by indirect** [comment]
Mirror separately sounds reasonable to me. 👍🏻

File: app/models/namespace/index/cooldown.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  performs def refresh
+    if versions = new_versions_since_last_refresh.includes(:gem).presence
+      manifest.append versions
+      touch
+    end
+  end
+
+  def new_versions_since_last_refresh
+    versions.published_after updated_at
```
Comment (indirect) [inline app/models/namespace/index/cooldown.rb]: I think we need a separate `compacted_at` so that unrelated changes can't make `updated_at` the wrong timestamp?

File: db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class AddLineToNamespaceGemVersions < ActiveRecord::Migration[8.1]
+  def change
+    add_column :namespace_gem_versions, :line, :string
+
+    up_only { Namespace::Gem::Version.includes(:references).find_each { _1.update line: _1.compute_line } }
+    StrongMigrations.disable_check :change_column_null_postgresql if defined?(StrongMigrations) # Shush, we're adding it in the same migration.
```
Comment (indirect) [inline db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb]: You can also wrap in `safety_guaranteed { }` 👍🏻

**Review by indirect — APPROVED** [review]
I like this approach, design-wise. 👍🏻

Do you have any thoughts about how we might be able to support cooldowns with different lengths? I'm not sure if we will want multiple on a single index (would someone want the public index but with 7 days? 14 days?) or if we will want the ability to adjust a single main cooldown (some company wants their own index to have a 5 day cooldown instead of the public 2 day cooldown?).

Both of those are really optional right now, but if you see a good way to support that based on this that would be cool.

File: db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb (hunk 2 above)
Comment (kaspth) [reply db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb]: Ah good call

File: app/models/namespace/index/cooldown.rb (hunk 1 above)
Comment (kaspth) [reply app/models/namespace/index/cooldown.rb]: Yeah maybe that's a good idea

Comment (kaspth) [reply app/models/namespace/index/cooldown.rb]: That'll be on the `Manifest` though.

**Comment by kaspth** [comment]
> Do you have any thoughts about how we might be able to support cooldowns with different lengths? I'm not sure if we will want multiple on a single index (would someone want the public index but with 7 days? 14 days?)

I've been simmering on this and I've got it working now, with a short explainer in https://github.com/gem-coop/peak/pull/197/commits/a33ec4eab6b9eb4d174e19156bc13f5a8dc4666a

```
gem.coop/@kaspth/dev/cooldown/7d # => find the 7 day cooldown
gem.coop/@kaspth/dev/cooldown/2d # => find the 2 day cooldown

gem.coop/@kaspth/dev/cooldown # => finds the first created cooldown
```

If you delete the implicit cooldown we'll correctly bust the cache and find the next one if any. We'll raise a 404 if there's no cooldown though.

> or if we will want the ability to adjust a single main cooldown (some company wants their own index to have a 5 day cooldown instead of the public 2 day cooldown?).

Yeah that's already supported, we just need to wire up some UI for it or allow it to be white glove for starters.

File: db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb (hunk 2 above)
Comment (kaspth) [reply db/migrate/20260428123819_add_line_to_namespace_gem_versions.rb]: Looks like it's `safety_assured`.

File: app/models/namespace/index/cooldown.rb (hunk 1 above)
Comment (kaspth) [reply app/models/namespace/index/cooldown.rb]: > I think we need a separate compacted_at so that unrelated changes can't make updated_at the wrong timestamp?

Ah actually this would be `last_refreshed_at`, and yeah let me look at that.

Comment (kaspth) [reply app/models/namespace/index/cooldown.rb]: Hitting some snags with this and I don't think I have the mental capacity to wrap this up after all.

Comment (kaspth) [reply app/models/namespace/index/cooldown.rb]: it's working now! I just forgot to add the column in a migration 😅
